### PR TITLE
ShyLU: Remove hwloc

### PR DESCRIPTION
**Pull request by sebrowne**
As far as I can tell, HWLOC is only used through Kokkos in ShyLU, so remove it as a TPL in the dependencies list.

@trilinos/shylu 

## Motivation
Want to be able to remove extra TPLs (especially in the Spack build, as per @rppawlo looking at our Spack package recently).

### CODE REVIEW

**Review by ndellingwood — APPROVED** [review]
This should be fine, in basker the usage in tests call out to public kokkos routines (defined regardless of hwloc availability)

File: packages/shylu/shylu_node/tacho/cmake/Dependencies.cmake [hunk 1]
```diff
@@ -3,6 +3,6 @@ SET(LIB_OPTIONAL_DEP_PACKAGES)
 SET(TEST_REQUIRED_DEP_PACKAGES Kokkos TrilinosSS)
 SET(TEST_OPTIONAL_DEP_PACKAGES)
 SET(LIB_REQUIRED_DEP_TPLS)
-SET(LIB_OPTIONAL_DEP_TPLS METIS HWLOC HYPRE MKL LAPACK BLAS Pthread QTHREAD VTune CUSOLVER CUSPARSE CUBLAS CUDA)
+SET(LIB_OPTIONAL_DEP_TPLS METIS HYPRE MKL LAPACK BLAS Pthread QTHREAD VTune CUSOLVER CUSPARSE CUBLAS CUDA)
```
[the remark this answers is not in the document]
Comment (sebrowne) [reply packages/shylu/shylu_node/tacho/cmake/Dependencies.cmake:6]: Do you want me to get rid of it in this same PR?  Happy to do so

**Comment by cgcgcg** [comment]
@sebrowne I removed hwloc and hypre in #12594. But I just realized that I forgot to also have the ShyLU/Basker change. Sorry about that. I'll take care of it.

**Comment by cgcgcg** [comment]
Taken care of in #12595.

**Pull request closed without merging**